src/Catel.MVVM/Catel.MVVM.Shared/MVVM/ViewModels/ViewModelBase.cs

```diff
@@ -1529,9 +1583,16 @@ protected virtual Task InitializeAsync()
 
             await OnClosedAsync(result);
 
-            Log.Info("Closed view model '{0}'", GetType());
+            var type = GetType();
+
+            Log.Info("Closed view model '{0}'", type);
 
             ViewModelManager.UnregisterViewModelInstance(this);
+
+            if (_reuseIdentifier)
```

Comment:
This code shouldn't be here. We should *always* release the identifiers. It's up to the *next* caller to decide whether the *next* vm should re-use this id. We should always release it.

Reply:
Yes

Reply:
So, Fixed